### PR TITLE
Address MariaDB limitation in strict mode

### LINKED ISSUE #6094: Database error during export of profile data
#### Description
Database Error: 'smf_home.pm.msgtime' isn't in GROUP BY

### Steps to reproduce
1.  Go to Profile - Download profile data - Check all checkboxes - Run export
2.  After some time we have an error in the Error log:

### Environment (complete as necessary)
- Version/Git revision:  6cda6d8e1c3b65ce80ab75de519e327b283196b2
- Database Type: MariaDB
- Database Version: 10.4.12
- PHP Version: 7.2.29

### PR DESCRIPTION
**Pull request by sbulen**
Fixes #6094 

@dragomano - please test!

@albertlast - input welcome.  I felt the more narrow query in the subquery made more sense.  I did A/B tests with my biggest users & no noticeable difference.

### CODE REVIEW

File: Sources/News.php [hunk 1]
```diff
@@ -2546,18 +2546,23 @@ function getXmlPMs($xml_format, $ascending = false)
...
+				INNER JOIN {db_prefix}pm_recipients AS pmr ON (pm2.id_pm = pmr.id_pm)
+				INNER JOIN {db_prefix}members AS mem ON (mem.id_member = pmr.id_member)
+			WHERE pm2.id_pm > {int:start_after}
+				AND (
+					(pm2.id_member_from = {int:uid} AND pm2.deleted_by_sender = {int:not_deleted})
+					OR (pmr.id_member = {int:uid} AND pmr.deleted = {int:not_deleted})
+				)
+			GROUP BY pm2.id_pm
+			LIMIT {int:limit} OFFSET {int:offset}
+		) nis ON nis.id_pm = pm.id_pm
```
Comment (jdarwood007) [inline Sources/News.php]: nitpick here but we have always used "AS" even though its not required, as we are verbose.  Should do that even for a sub select.

**Comment by dragomano** [comment]
It works fine, thanks :)

**Comment by albertlast** [comment]
my test env didn't got a big pm setup.. so i cant this from a performance pov.
Personaly i had tried before to solve this by using distinct
```
	$request = $smcFunc['db_query']('', '
		SELECT DISTINCT pm.id_pm, pm.msgtime, pm.subject, pm.body, pm.id_member_from, pm.from_name, ' . $select_id_members_to . ' AS id_members_to, ' . $select_to_names . ' AS to_names
		FROM {db_prefix}personal_messages AS pm
			INNER JOIN {db_prefix}pm_recipients AS pmr ON (pm.id_pm = pmr.id_pm)
			INNER JOIN {db_prefix}members AS mem ON (mem.id_member = pmr.id_member)
		WHERE pm.id_pm > {int:start_after}
			AND (
				(pm.id_member_from = {int:uid} AND pm.deleted_by_sender = {int:not_deleted})
				OR (pmr.id_member = {int:uid} AND pmr.deleted = {int:not_deleted})
			)
		ORDER BY pm.id_pm {raw:ascdesc}
		LIMIT {int:limit} OFFSET {int:offset}',

**Comment by sbulen** [comment]
It performs well on my test copy of my prod forum.   

I think it's either COUNT DISTINCT *or* GROUP_CONCAT...  but we need the GROUP_CONCAT function, so distinct doesn't work:
https://dev.mysql.com/doc/refman/8.0/en/group-by-functions.html

It works in my (small) pg environment.

**Comment by albertlast** [comment]
Okay didn't notice the group_concat...
from a technical pov is the subqery from you differently to orginal because of position of the order by.

Limit is affected by the order by,
so to spread them into different parts of the query could lead in a different result.

**Comment by sbulen** [comment]
Correct.  GROUP BY doesnt necessarily sort.  I'll tweak it a bit tomorrow...

**Comment by sbulen** [comment]
*** Note I also removed the OFFSET for posts & PMs...  ***

The offset should be a #, a count of msgs, _usually a multiple of your LIMIT_.  Note that the logic saved off the last msg or pm retrieved, and subsequent calls included that in the WHERE clause.  So LIMIT by itself is appropriate here.  

Input welcome.

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]
I'm fine with removing the use of `OFFSET` from these queries, sure. I added that back in my first version of the export feature, but then I switched over to using `WHERE pm.id_pm > {int:start_after}` for greater efficiency, which means we no longer using the offset value at all.

However, if you remove `OFFSET` from these queries, you also need to remove all occurrences of `$context['xmlnews_offset']` in all files.

**Comment by sbulen** [comment]
Done, & tests OK.  

I thought that was used by the xml feeds prior to #5081 ...  Interesting, it was defined, just never used.

**Comment by sbulen** [comment]
@Sesquipedalian - if this looks good let me know, we can merge

**Review by Sesquipedalian — APPROVED** [review]

**Pull request merged**